plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/registry/ISQLPresentationContributor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ui.editors.sql.registry;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Represents a contributor to the SQL editor presentation system.
+ *
+ * <p>This interface is used to describe optional editor presentations (e.g. Visual Query Builder)
+ * that can be enabled or disabled via preferences. Implementations of this interface provide metadata
+ * for the UI (labels, tooltips), preference integration (setting key), and dynamic switching logic.</p>
+ *
+ * <p>These contributors are typically declared via extension points in plugin.xml and consumed
+ * by {@link SQLPresentationRegistry} to build the presentation list and the preferences page.</p>
+ */
+public interface ISQLPresentationContributor {
```

Comment:
Why adding new interface? I think we can use existing presentation descriptor with 1 extra attribute (enabledWhen).